### ISSUE TITLE
How to conditionally render the button in react-bootstrap-table ? (The button for each row)

### ISSUE THREAD

**Issue by aziz-boudi4** [AllenFang/react-bootstrap-table#1201]
I'm trying to conditionally render the button in the `react-bootstrap-table` by comparing the row.id with the item.id from my database (firebase)

- I'm using the table to display groups that I get from the meetup-api.
- once I fetch all the groups, I compare their ids (row.id) with the groups that I have on my database.
- If they match I display Button1
- If they don't match I display Button2

I tried many attempts but my solutions are not giving me the desired result 

Here's my code : 

This Solution is working only for the latest pressed button, but I want to get the real representation
If I already have 8 groups in the database, the buttons of the 8 groups should be different than the other buttons
    
```javascript
    class App extends Component {
      
    constructor() {
      super()
      this.state = {
         json: [], // json(coming from the Meetup-api)
         jsonFromDatabase: [],
      }
      this.cellButton = this.cellButton.bind(this);
    }
      
    cellButton(cell, row, enumObject, rowIndex) {
      let theButton
      for(var group in this.state.jsonFromDatabase){
      if (this.state.jsonFromDatabase[group].id !== row.id){
         // Display this button if the group is not in the database
         theButton = <button style={{ backgroundColor: "blue"}}
                         type="button"
                         onClick={() => this.onClickGroupSelected(cell, row, rowIndex)}>
                       Process the group
                     </button>
       } else {
         // Display this button if the group is already in the database
         theButton = <button style={{ backgroundColor: "red" }}
                         type="button"
                         onClick={() => this.onClickGroupToUpdate(cell, row, rowIndex)}>
                       Update the group
                     </button>
         }
      }
      return theButton
    }

    render() {
        return (
          <BootstrapTable data={this.state.json} options={ this.options } >
              <TableHeaderColumn isKey dataField='id' width='100'>ID</TableHeaderColumn>
              <TableHeaderColumn dataField='name' width='300'>Group Name</TableHeaderColumn>
              <TableHeaderColumn dataField='button' width='100' dataFormat={this.cellButton}>Generate Group Page</TableHeaderColumn>
           </BootstrapTable>
        )
      }
    }
```

Another unsuccessful variation that I tried:

```javascript
    cellButton(cell, row, enumObject, rowIndex) {
        return this.state.jsonFromDatabase.map((group, i) => {
            if (group.id !== row.id){
                return  <button style={{ backgroundColor: "blue"}}
                                   type="button"
                                   onClick={() => this.onClickGroupSelected(cell, row, rowIndex)}>
                                        Process the group
                         </button>
            }else{
               return  <button style={{ backgroundColor: "red" }}
                                  type="button"
                                  onClick={() => this.onClickGroupToUpdate(cell, row, rowIndex)}>
                                         Update the group
                         </button>
            }
       })
  }
```
    

![screen shot 2017-04-05 at 7 58 36 pm](https://cloud.githubusercontent.com/assets/7435015/24736113/057a1550-1a3b-11e7-93c6-f2916155daf4.png)

Here is s a link to my Stackoverflow question: http://stackoverflow.com/questions/43244914/how-to-conditionally-render-the-button-in-react-bootstrap-table-the-button-fo

**Comment by AllenFang** [comment]
> This Solution is working only for the latest pressed button, but I want to get the real representation
> If I already have 8 groups in the database, the buttons of the 8 groups should be different than the other buttons

Sorry I still get some confuse on here, but I thought you code is good, the `dataFormat` can help you to custom the cell and I dont know what's problem you encounter, is that related with `react-bootstrap-table`?

Thanks 👍

**Comment by aziz-boudi4** [comment]
Thanks for the quick reply @AllenFang , I'll try to be more precise

- I have an array of objects in this.state.json (that holds data from the api call)
- and I have another array of objects in this.state.jsonFromMyDatabase (the data already in my database)

inside the cellButton function : 
- I want to compare the objects in each array and if some objects have a similar `id` I display a `red` button otherwise I display a `blue`button.
- basically if it's already in my database it will have a different button

My questions are: 
1- what does dataFormat expect to receive ? does it expect to receive an Array of buttons ?

**Comment by aziz-boudi4** [comment]
I have an object of subObjects in my database like this : 
```javascript
databaseObject : { 
   subObj1 : { "id": "000001", name: "testName1", city: "cityTest1" },
   subObj2: { "id": "000002", name: "testName2", city: "cityTest2" },
   subObj3: { "id": "000003", name: "testName3", city: "cityTest3" }
}
```

Can I compare the `ids`of the `subObj` with the each row.id   inside the function that will go inside dataFormat , or should I do it somewhere else : 

The idea is to do something like :
looping through the dabaseObject and compare the `id` of each `subObject` with the the id of the row that will be displayed in the table 

```javascript
cellButton(cell, row, enumObject, rowIndex) {
   if (row.id !== subObj.id) {
       return btn1
   } else {
      return btn2
   } 
}
````

**Comment by aziz-boudi4** [comment]
Ok, I managed to find a solution to conditionally display a different button depending on whether the  row.id is stored in the database or not   

here's the code:
```javascript
  cellButton(cell, row, enumObject, rowIndex) {
    let theButton;
    let groupExistsInDatabase = false
    for (var group in this.state. jsonFromDatabase){
      if (this.state. jsonFromDatabase[group].id === row.id){
        groupExistsInDatabase = true
        break;
      }
    }

    if (groupExistsInDatabase === true){
      theButton =  <button style={{ backgroundColor: "red"}}
                  type="button"
                  onClick={() => this.onClickGroupToUpdate(cell, row, rowIndex)}>
                  Update in Asana
              </button>
    } else {
      theButton =  <button style={{ backgroundColor: "blue" }}
                  type="button"
                  onClick={() => this.onClickGroupSelected(cell, row, rowIndex)}>
                  Send to Asana
              </button>
    }
    return theButton
  }  
```
